lib/shopify_api/resources/ping/message.rb

```diff
@@ -1,9 +1,23 @@
 # frozen_string_literal: true
 
+require 'byebug'
+
 module ShopifyAPI
   module Ping
     class Message < Base
       self.prefix = "/admin/api/ping-api/v1/conversations/:conversation_id/"
+
+      def successful_delivery(options)
+        delivery_details = ShopifyAPI::Ping::DeliveryConfirmation.new(options)
+        delivery_details.save
+        "true"
+      end
+
+      def failed_delivery(options)
```

Comment:
the content of this method seems similar to the above. shouldn't this method build the body for a failed message confirmation and the above method have the body for a successful message?

Reply:
Then are we looking to create the entire `delivery_confirmation_details` hash within each of these methods? I thought we were just passing whether the message was successfully delivered, and if not, the string containing the error message. And then move the responsibility of processing the response to the `ping-api` side.